### PR TITLE
Euler's Totient Function - Java

### PR DESCRIPTION
**Pull request by SangeetaMishra**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#445

### CODE REVIEW

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@SangeetaMishra LGTM! Format the code like this
```java
/*
Euler's totient function, also known as phi-function ϕ(n), counts the
numberof integers between 1 and n inclusive, which are coprime to n.
*/
import java.util.Scanner;

public class Euler_Totient {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter an integer: ");
		int n = sc.nextInt();

		int temp = n;
		//This gives euler totient function for N
		// Time Complexity: O(N*root(N))
		int ans = n;
		for (int i = 2; i <= n; ++i) {
			if (n % i == 0) {
				ans = ans - ans / i;
				while (n % i == 0)
					n /= i;
			}
		}
		if (n > 1) {
			ans = ans - ans / n;
		}
		n = temp;
		System.out.println("\nThe euler totient function for " + n + " is " + ans);

		//This gives euler totient function from 1 to N
		// Time Complexity: O(N loglogN) - same as Sieve of Eratosthenes
		int phi[] = new int[n + 1];
		for (int i = 0; i <= n; ++i) {
			phi[i] = i;
		}
		for (int i = 2; i <= n; ++i) {
			if (phi[i] == i) {
				for (int j = i; j <= n; j += i) {
					phi[j] = phi[j] - phi[j] / i;
				}
			}
		}
		System.out.println("\nThe euler totient function for all integers from 1 to " + n + " is:\n");
		for (int i = 1; i <= n; ++i) {
			System.out.println(i + " " + phi[i]);
		}
	}
}

/*
Time Complexity: O(N loglogN)
OUTPUT
Enter an integer: 10
The euler totient function for 10 is: 4
The euler totient function for all integers from 1 to 10 is:
1: 1
2: 1
3: 2
4: 2
5: 4
6: 2
7: 6
8: 4
9: 6
10: 4
-----
(1,3,7,9) are coprime to 10
*/
```

**Review by iamrajiv — APPROVED** [review]

File: Java/math/Euler's_Totient_function.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+			}
+		}
+		if (n > 1) {
+			ans = ans - ans / n;
+		}
+		n = temp;
+		System.out.println("\nThe euler totient function for " + n + " is " + ans);
+
+		//This gives euler totient function from 1 to N
+		// Time Complexity: O(N loglogN) - same as Sieve of Eratosthenes
```
Comment (Kajol-Kumari) [inline Java/math/Euler's_Totient_function.java:31]: can we please remove this line as we have already included it in the commented form below the code section.

**Review by Kajol-Kumari — APPROVED** [review]
@SangeetaMishra Minor change required, Rest LGTM :+1:

File: Java/README.md [hunk 2]
```diff
@@ -75,9 +75,10 @@ _add list here_
 ## Mathematics
 
 - [Count Number Of Digits](math/countDigits)
-  - [Iterative](countDigits/iterative.java)
-  - [Recursive](countDigits/recursive.java)
-  - [String Method](countDigits/stringmethod.java)
+- [Euler's Totient Function](math/Euler's_Totient_function.java)
+- [Iterative](countDigits/iterative.java)
```
Comment (atarax665) [inline Java/README.md]:
```suggestion
- [Iterative](math/countDigits/iterative.java)
```

File: Java/README.md [hunk 3]
```diff
@@ -75,9 +75,10 @@ _add list here_
...
+- [Recursive](countDigits/recursive.java)
```
Comment (atarax665) [inline Java/README.md]:
```suggestion
- [Recursive](math/countDigits/recursive.java)
```

File: Java/README.md [hunk 4]
```diff
@@ -75,9 +75,10 @@ _add list here_
...
+- [String Method](countDigits/stringmethod.java)
```
Comment (atarax665) [inline Java/README.md]:
```suggestion
- [String Method](math/countDigits/stringmethod.java)
```

**Review by atarax665 — APPROVED** [review]
Make these changes so that I can merge this PR.

**Comment by SangeetaMishra** [comment]
@atarax665 I have made the required changes

**Pull request merged**